### ISSUE TITLE
F6.2 center check does not consider fab lines

### ISSUE THREAD

**Issue by evanshultz** [KiCad/kicad-library-utils#242]
https://github.com/KiCad/kicad-footprints/pull/570 has an issue with centering horizontally, according to Travis, with the footprint center at `(-8.025,0.0)mm`. I verified with the current HEAD scripts run locally.

But actually, there is no problem. The extents of the pads on the left and fab lines on the right are +/-10.9mm.

I don't see anything obviously wrong with the footprint. A few tests:
- If I move the fab lines on the right side leftward, the calculated center doesn't change.
- If I move the two pad 5s, the calculated center does change.

So it seems the scripts are not taking into account the fab lines when calculating the footprint center.

**Comment by Ratfink** [comment]
Yes, it only calculates the center based on the pads' shapes, which happens on [line 27 of pcb/rules/F6_2.py](https://github.com/KiCad/kicad-library-utils/blob/master/pcb/rules/F6_2.py#L27).  This works for many packages, but as you point out it really isn't the right approach overall.  It would be better to consider the fab layer lines as well.

**Comment by cpresser** [comment]
Actually there is already a proposed patch for that.
https://github.com/KiCad/kicad-library-utils/pull/260
